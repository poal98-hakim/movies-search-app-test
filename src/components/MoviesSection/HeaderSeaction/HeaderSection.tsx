import React, { useContext } from "react";
import RightArrowIcon from "../../../images/right-arrow.png";
import LeftArrowIcon from "../../../images/left-arrow.png";
import MainContext from "../../../contexts/MainContext";
import ImageButton from "./ImageButton/ImageButton";

interface IProps{
  hasMore: boolean;
}

const HeaderSection : React.VFC<IProps> = ({ hasMore }) => {
  const {page, setPage} = useContext(MainContext);

  return (
    <div className="flex f-justify-end">
        <ImageButton 
            icon={{src: LeftArrowIcon, alt: "left-arrow"}} 
            onClick={() => setPage(page => page-1)} 
            disabled={page ? false : true}
            />
        <ImageButton 
            icon={{src: RightArrowIcon, alt: "right-arrow" }} 
            onClick={() => setPage(page => page+1)} 
            disabled={!hasMore}
            />
    </div>
  )
}
export default HeaderSection;