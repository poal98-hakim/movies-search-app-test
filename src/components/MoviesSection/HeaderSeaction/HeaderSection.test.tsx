import HeaderSection from "./HeaderSection";
import { render, screen } from '@testing-library/react';

const PROPS = {
    hasMore: true
}

describe('HeaderSection', () => {
  let props = {...PROPS};

  beforeEach(() => { //before each test render the component and reset the value of props
    props = {...PROPS};
    render(<HeaderSection {...props} />);
  });
  
  it('renders correctly', () => {
    const { baseElement } = render(<HeaderSection {...props} />);
    expect(baseElement).toMatchSnapshot()
  })

  it('loads buttons', () => {
    expect(screen.getAllByRole("button")).toHaveLength(2);
  })

  it('shows correctly icons', () => {
    expect(screen.getByAltText("left-arrow")).toBeInTheDocument();
    expect(screen.getByAltText("right-arrow")).toBeInTheDocument();
    expect(screen.getAllByRole("img")[0]).toHaveAttribute("src", expect.stringContaining("left-arrow.png"));
    expect(screen.getAllByRole("img")[1]).toHaveAttribute("src", expect.stringContaining("right-arrow.png"));
  })

})
