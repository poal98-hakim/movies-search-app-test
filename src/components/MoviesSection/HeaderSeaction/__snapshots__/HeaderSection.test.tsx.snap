// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderSection renders correctly 1`] = `
<body>
  <div>
    <div
      class="flex f-justify-end"
    >
      <button
        class="image-button"
        disabled=""
      >
        <img
          alt="left-arrow"
          src="left-arrow.png"
          width="25"
        />
      </button>
      <button
        class="image-button"
      >
        <img
          alt="right-arrow"
          src="right-arrow.png"
          width="25"
        />
      </button>
    </div>
  </div>
  <div>
    <div
      class="flex f-justify-end"
    >
      <button
        class="image-button"
        disabled=""
      >
        <img
          alt="left-arrow"
          src="left-arrow.png"
          width="25"
        />
      </button>
      <button
        class="image-button"
      >
        <img
          alt="right-arrow"
          src="right-arrow.png"
          width="25"
        />
      </button>
    </div>
  </div>
</body>
`;
